### PR TITLE
[ZK-152] Execution state gadget for EcAdd precompile call

### PR DESCRIPTION
**Pull request by pjxthy**
Changes:
- Merge PR difference: https://github.com/scroll-tech/zkevm-circuits/pull/598

Test Plan:
- Green CIs

TODOs:
- May need @akloopx 's help to disable labeler for the failing test: https://deltafi-ai.atlassian.net/browse/ZK-155

### CODE REVIEW

File: bus-mapping/src/evm/opcodes/callop.rs [hunk 1]
```diff
@@ -380,39 +381,37 @@ impl<const N_ARGS: usize> Opcode for CallOpcode<N_ARGS> {
...
-                                log_id: None,
-                                rw_counter_start,
-                                copy_bytes: CopyBytes::new(copy_steps, None, Some(prev_bytes)),
-                            },
-                        );
-                        Some(output_bytes)
-                    } else {
-                        None
-                    };
+                let output_bytes = if call.is_success() && !result.is_empty() {
```
Comment (akloopx) [inline bus-mapping/src/evm/opcodes/callop.rs:384]: why remove all 'call.call_data_length > 0' condition?

File: bus-mapping/src/evm/opcodes/callop.rs [hunk 2]
```diff
@@ -380,39 +381,37 @@ impl<const N_ARGS: usize> Opcode for CallOpcode<N_ARGS> {
...
+                    Some(output_bytes)
+                } else {
+                    None
+                };
 
                 // insert another copy event (output) for this step.
                 let rw_counter_start = state.block_ctx.rwc;
-                let returned_bytes = if call.is_success() && call.call_data_length > 0 && length > 0
-                {
+                let returned_bytes = if call.is_success() && length > 0 {
```
Comment (akloopx) [inline bus-mapping/src/evm/opcodes/callop.rs:414]: ditto

**Review by akloopx — CHANGES_REQUESTED** [review]

File: zkevm-circuits/src/evm_circuit/util/precompile_gadget.rs [hunk 3]
```diff
@@ -162,7 +154,38 @@ impl<F: Field> PrecompileGadget<F> {
                 });
             }),
             Box::new(|_cb| { /* Modexp */ }),
-            Box::new(|_cb| { /* Bn128Add */ }),
```
Comment (RuiLiu3) [inline zkevm-circuits/src/evm_circuit/util/precompile_gadget.rs:165]: What does Bn128Add mean? is the new add code to implement the logic about it?

**Review by RuiLiu3 — APPROVED** [review]

File: zkevm-circuits/src/evm_circuit/util/precompile_gadget.rs (hunk 3 above)
Comment (dfixx) [reply zkevm-circuits/src/evm_circuit/util/precompile_gadget.rs:165]: Bn128Add是一种Precompile方式，后面关联着stateExecute，这个跟具体的EVM里的Cell有关系，也跟gas有关系，但是这个具体的操作意思，我还没看懂。。。

**Pull request merged**